File: linked-list-cycle/linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        if(!head) return false;
        ListNode *tor = head, *hare = head;
        
        while(hare and hare->next)
        {
            tor = tor->next;
            hare = hare->next->next;
            
            if(tor == hare) return true;
        }
        
        return false;
    }
};

File: linked-list-cycle/linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        if(head == null) return false;
        ListNode tor = head , hare = head;
        while(hare != null && hare.next != null)
        {
            hare = hare.next.next;
            tor = tor.next;
            
            if(hare == tor) return true;
        }
        
        return false;
    }
}